source/initial_temperature/S40RTS_perturbation.cc

```diff
@@ -337,8 +340,28 @@ namespace aspect
 
       double temperature_perturbation;
       if (depth > no_perturbation_depth)
-        // scale the density perturbation into a temperature perturbation
-        temperature_perturbation =  -1./thermal_alpha * density_perturbation;
+        {
+          // scale the density perturbation into a temperature perturbation
+          // see if we need to ask material model for the thermal expansion coefficient
+          if (use_material_model_thermal_alpha)
+            {
+              MaterialModel::MaterialModelInputs<3> in(1, this->n_compositional_fields());
+              MaterialModel::MaterialModelOutputs<3> out(1, this->n_compositional_fields());
+              in.position[0] = position;
+              in.temperature[0] = background_temperature;
+              in.pressure[0] = this->get_adiabatic_conditions().pressure(position);
```

Comment:
Is it consistent to use the pressure from the adiabatic profile, but the background temperature for the temperature to compute the thermal expansivity? I do not think it matters much as thermal_expansivity is not often set to be temperature depedent (although in theory it is), but have you thought about which one to choose, and do you think this is the better choice?

Reply:
I copied the MaterialModelInputs initialization from another file, changed background_temperature, and did not think about that - I'll get back to you.

Reply:
It looks like `background_temperature` is set to the adiabatic temperature in compressible models and the reference temperature in incompressible models. That sounds reasonable, unless maybe someone uses an incompressible material model with temperature dependent thermal expansion (which sounds unlikely).

Reply:
Ok, sounds reasonable. I will let @jaustermann push the merge button.